signing/src/main/java/tech/pegasys/web3signer/signing/secp256k1/azure/AzureConnectionFactory.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2023 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.signing.secp256k1.azure;
+
+import tech.pegasys.web3signer.keystorage.azure.AzureConnection;
+import tech.pegasys.web3signer.keystorage.azure.AzureConnectionParameters;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.time.Duration;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class AzureConnectionFactory {
+  private final Map<URI, HttpClient> httpClientMap = new ConcurrentHashMap<>();
```

Comment:
Does the entries here need to be cleaned up at all or is it OK to leave them there for the lifetime of the app?

Reply:
If we kill the connection we lose the performance gain of having that cached. Which basically drops the signing time significantly overall.

Reply:
What determines the URI key, is it effectively a single entry map (one azure url) or is there one per key, etc?

Reply:
one per vault

Reply:
I think this is the same comment/resolution: https://github.com/Consensys/web3signer/pull/862#discussion_r1277151220